modules/linx.py

```diff
@@ -15,26 +15,22 @@ def linx(phenny, input, short=False):
 
     url = input.group(2)
     if not url:
-        phenny.reply("No URL provided. CAN I HAS?")
+        phenny.reply("No URL provided")
         return
 
     try:
-        req = web.post("https://linx.li/upload/remote", {'url': url, 'short': short, 'api_key': phenny.config.linx_api_key})
-    except (web.HTTPError, web.ConnectionError):
-        raise GrumbleError("Couldn't reach linx.li")
 
-    data = json.loads(req)
-    if len(data) <= 0 or not data['success']:
-        phenny.reply('Sorry, upload failed.')
-        return
+        url = url.replace(".onion/", ".onion.to/")

```

Comment:
I'd rather not do this. I understand the benefit, but I don't really like relying on random tor2web gateways.

An idea I've had for a while is the ability to configure phenny so that outgoing HTTP/HTTPS requests go over a proxy that can be configured separately than the proxy used to connect to IRC. Now that requests has SOCKS support, this shouldn't be particularly difficult. I'd prefer telling people to do that, rather than using a tor2web gateway. Perhaps the configuration could even work on a per-module basis.